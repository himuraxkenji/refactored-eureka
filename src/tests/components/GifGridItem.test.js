import '@testing-library/jest-dom';
import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Prueba en <GrifGridItem />', () => {

  const title = 'Soy un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(
    <GifGridItem
      title={title}
      url={url}
    />
  );


  test('debe mostrar el componente correctamente', () => {

    expect(wrapper).toMatchSnapshot();

  });


  test('debe tener un parrafo con el title', () => {

    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);

  });


  test('debe de tener la imagen igual al url y alt de los props', () => {

    const img = wrapper.find('img');
    const src = img.prop('src');
    const alt = img.prop('alt');

    expect(src).toBe(url);
    expect(alt).toBe(title);

  });

  test('debe de tener animate__fadeIn', () => {

    const div = wrapper.find('div');
    const animation = div.prop('className');
    const isAnimated = animation.includes('animate__fadeIn');

    expect(isAnimated).toBe(true);

  });

}); 
