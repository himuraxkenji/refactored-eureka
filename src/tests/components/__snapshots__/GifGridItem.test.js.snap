// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GrifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Soy un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Soy un titulo
  </p>
</div>
`;
